### PR TITLE
Docs for redacted vars

### PR DESCRIPTION
**Pull request by plaindocs**
Docs for https://github.com/buildkite/agent/pull/1109

I think this covers the basics.

### CODE REVIEW

**Review by nikwek — COMMENTED** [review]
in pipelines/secres, why not mention `*_SECRET` and `*_TOKEN` as well? That would be consistent with the other pages. 

in pipelines/managing_log_output (this now real nit-picking.. sorry) I noticed the font was different for the env variable and then realized it was because it was a link... Am I being too picky? This would avoid that: 

You can replace the default patterns by [setting the Buildktie Agent’s optional attribute](/docs/agent/v3/configuration#redacted-vars) `BUILDKITE_REDACTED_VARS`.

File: pages/pipelines/secrets.md.erb [hunk 1]
```diff
@@ -128,6 +128,8 @@ steps:
 
 Referencing secrets in your steps risks them being interpolated, uploaded to Buildkite, and shown in plain text in the "Uploaded Pipelines" list in the job’s Timeline tab.
 
+The Buildkite agent does [redact environment variables](/docs/pipelines/managing-log-output#redacted-environment-variables) that match common password patterns such as `*_PASSWORD`.
```
Comment (keithpitt) [inline pages/pipelines/secrets.md.erb:131]: Perhaps `[redact values of environment variables]`?

File: pages/pipelines/managing_log_output.md.erb [hunk 2]
```diff
@@ -110,3 +110,14 @@ xcodebuild <build options> | tee -a build.log | xcpretty -c
 Make sure to set the `-o pipefail` option in your buildscript as above, otherwise the build failure status might not be passed through correctly.
 
 <%= image("xcpretty.png", alt: "Screenshot of xcpretty output", size: '656x316') %>
+
+## Redacted environment variables
+
+Environment variables matching common patterns for passwords and other secure information are redacted before the build log is uploaded to Buildkite.
```
Comment (keithpitt) [inline pages/pipelines/managing_log_output.md.erb:116]: I can't decide if "Environment variables" or "Values of environment variables" are basically the same thing

Comment (keithpitt) [reply pages/pipelines/managing_log_output.md.erb:116]: "Values of environment variables who's names match common patterns" or something?

File: pages/pipelines/managing_log_output.md.erb [hunk 3]
```diff
@@ -110,3 +110,14 @@ xcodebuild <build options> | tee -a build.log | xcpretty -c
...
+
+The default patterns are `*_PASSWORD`, `*_SECRET` and `*_TOKEN`. You can replace the default patterns by [setting `BUILDKITE_REDACTED_VARS`](/docs/agent/v3/configuration#redacted-vars) *on your agents*. 
+
+<div class="Docs__note">
+<p class="Docs__note__heading">Setting environment variables</p>
+<p>Note that if you **set** an environment variable in your <code>pipeline.yml</code> it is not redacted, but doing that is <a href="/docs/pipelines/secrets#anti-pattern-storing-secrets-in-your-pipeline-dot-yml">not recommended</a>.</p>
```
Comment (keithpitt) [inline pages/pipelines/managing_log_output.md.erb:122]: Perhaps need to call out that setting a "secret" is bad? This kinda reads like setting any environment variable is not recommended

File: pages/pipelines/managing_log_output.md.erb [hunk 4]
```diff
@@ -110,3 +110,21 @@ xcodebuild <build options> | tee -a build.log | xcpretty -c
...
+
+The Buildkite agent can redact strings that match the value of environment variables whose names match common patterns for passwords and other secure information before the build log is uploaded to Buildkite.
+
+The default environment variable name patterns are `*_PASSWORD`, `*_SECRET` and `*_TOKEN`. You can replace the default patterns by [setting the environment variable](/docs/agent/v3/configuration#redacted-vars) `BUILDKITE_REDACTED_VARS` **on your agent**.
+
+For example, if you have environmment variable `MY_SECRET="topsecret"`and you run a command that outputs `This is topsecret info`, the log output will actually be `This is [REDACTED] info`.
+
+<div class="Docs__note">
+<p class="Docs__note__heading">Setting environment variables</p>
+<p>Note that if you <emphasis>set</emphasis> a secret environment variable in your <code>pipeline.yml</code> it is not redacted, but doing that is <a href="/docs/pipelines/secrets#anti-pattern-storing-secrets-in-your-pipeline-dot-yml">not recommended</a>. Also if an environment variable is present in a Buildkite command step it will not be redacted.</p>
```
Comment (sj26) [inline pages/pipelines/managing_log_output.md.erb:129]: I think these are the same thing — it appears in the command if it is interpolated into the command within the pipeline.yml.

```suggestion
<p>Note that if you <emphasis>set</emphasis> or <emphasis>interpolate</emphasis> a secret environment variable in your <code>pipeline.yml</code> it is not redacted, but doing that is <a href="/docs/pipelines/secrets#anti-pattern-storing-secrets-in-your-pipeline-dot-yml">not recommended</a>.</p>
```

Comment (plaindocs) [reply pages/pipelines/managing_log_output.md.erb:129]: Much better, cheers

File: pages/agent/v3/configuration.md.erb [hunk 5]
```diff
@@ -343,6 +343,15 @@ _Optional attributes:_
       </td>
     </tr>
 
+    <tr id="redacted-vars">
+      <th><code>redacted-vars</code></th>
+      <td>
+        Strings matching the values of environment variables whose names match these pattern are <a href="/docs/pipelines/managing-log-output#redacted-environment-variables">redacted</a> before being printed to the build log. This is an experimental feature, set <code>experiment="redacted-vars"</code> in your <a href="/docs/agent/v3/configuration#experiment"> agent configuration</a> to use it.
```
Comment (keithpitt) [inline pages/agent/v3/configuration.md.erb:349]:
```suggestion
        Strings matching the values of environment variables whose names match these patterns are <a href="/docs/pipelines/managing-log-output#redacted-environment-variables">redacted</a> before being printed to the build log. This is an experimental feature, set <code>experiment="redacted-vars"</code> in your <a href="/docs/agent/v3/configuration#experiment"> agent configuration</a> to use it.
```

**Review by keithpitt — APPROVED** [review]

File: pages/pipelines/managing_log_output.md.erb [hunk 6]
```diff
@@ -110,3 +110,21 @@ xcodebuild <build options> | tee -a build.log | xcpretty -c
 Make sure to set the `-o pipefail` option in your buildscript as above, otherwise the build failure status might not be passed through correctly.
 
 <%= image("xcpretty.png", alt: "Screenshot of xcpretty output", size: '656x316') %>
+
+## Redacted environment variables
+
+<div class="Docs__experiment">
+<p class="Docs__experiment__heading">Experimental feature</p>
+<p>This is an experimental feature, set <code>experiment="redacted-vars"</code> in your <a href="/docs/agent/v3/configuration#experiment"> agent configuration</a> to use it.</p>
```
Comment (pda) [inline pages/pipelines/managing_log_output.md.erb:118]: @plaindocs This should be `experiment="output-redactor"`… `redacted-vars` is a separate flag for changing which env var values should be redacted.

Comment (pda) [reply pages/pipelines/managing_log_output.md.erb:118]: PR at https://github.com/buildkite/docs/pull/978

**Pull request merged**